pkg/network/dns_parser.go

```diff
@@ -14,6 +14,12 @@ const maxIPBufferSize = 200
 var (
 	errTruncated      = errors.New("the packet is truncated")
 	errSkippedPayload = errors.New("the packet does not contain relevant DNS response")
+
+	// recordedRecordTypes defines a map of DNS types that we'd like to capture.
+	// add additional types here to add DNSQueryTypes that will be recorded
+	recordedQueryTypes = map[layers.DNSType]struct{}{
+		layers.DNSTypeA:    {},
+		layers.DNSTypeAAAA: {}}
```

Comment:
We should not store stats for this new type (AAAA) until we have fixed memory issues related to it.

Reply:
@mbakht We are going to merge this PR as-is, while we work on the memory usage. If we are unable to improve the memory usage before the code freeze, we will remove `AAAA`. This is in addition to adding a configuration option to control this list (in a separate PR).

Reply:
Even if we can improve the memory usage, IMHO we should first see the impact of turning on domain stats collection for our clients. Turning on domain collection and and adding processing of one more query type in the **same** release would make it hard to evaluate the impact of each of them separately. Like all other features that can affect performance, I think we should enable `AAAA` first in our prod and then, depending on the impact, turn it on for our clients.

Reply:
> I think we should enable AAAA first in our prod and then, depending on the impact, turn it on for our clients.

We should be able to evaluate the performance of system-probe without waiting for an entire agent release cycle. Adding query types, and adding collection of `AAAA` queries, does not change the memory usage semantics. It is all based on the number of requests captured. 

Per-domain collection does change the semantics since it can cause previously aggregated stats to be split. 

> Turning on domain collection and and adding processing of one more query type in the same release would make it hard to evaluate the impact of each of them separately.

Query type and per-domain do not have to be mutually turned on. You might want to capture stats for `AAAA` queries, while also not collecting per-domain stats.